frontend/mr-admin-flate/src/components/redaksjoneltInnhold/AmoKategoriseringSchema.ts

```diff
@@ -2,15 +2,25 @@ import { Bransje, ForerkortKlasse, InnholdElement, Sertifisering } from "@mr/api
 import z from "zod";
 
 const InnholdElementerSchema = z
-  .nativeEnum(InnholdElement, { required_error: "Du må velge minst ett element" })
+  .nativeEnum(InnholdElement, { errorMap: () => ({ message: "Du må velge minst ett element" })})
   .array()
   .nonempty("Du må velge minst ett element");
 
 export const AmoKategoriseringSchema = z.discriminatedUnion("kurstype", [
   z.object({
     kurstype: z.literal("BRANSJE_OG_YRKESRETTET"),
-    bransje: z.nativeEnum(Bransje, { required_error: "Du må velge bransje" }),
-    sertifiseringer: z.custom<Sertifisering>().array().default([]),
+    bransje: z.nativeEnum(Bransje, { errorMap: () => ({ message: "Du må velge bransje" })}),
+    sertifiseringer: z
+          .custom<Sertifisering>()
+          .array()
+          .nullish()
+          .transform((val) => {
+            if (!val) {
+                return [];
+            }
+            return val;
+          })
+          .pipe(z.custom<Sertifisering>().array()),
```

Comment:
Bedre måte å gjøre dette på? Man kunne satt default variabler når kurstypen endret seg, men det er ikke så gøy heller. Transform skjer etter validering, så man får valideringsfeil når man endrer kurstype og denne blir null. ~Som jeg trodde `.default([])` skulle egentlig fikse.~ Hmm det er nok fordi ControlledMultiSelect setter verdien til null, og default bare kjører når verdien er undefined. Skal se om jeg kan fikse

Merkelig at jeg ikke trenger det for `forerkort` under her... Må være en forskjell på aksel sin mulitselect og vår.

Reply:
Hmm det er nok fordi ControlledMultiSelect setter verdien til null, og default bare kjører når verdien er undefined. Skal se om jeg kan fikse